.github/test-matrix.json

```diff
@@ -0,0 +1,110 @@
+{
```

Comment:
How was this file built? It seems brittle and prone to get stale as new modules are added and dependencies change

Reply:
It replaces the list that's here: https://github.com/trinodb/trino/blob/master/.github/workflows/ci.yml#L336

I guess I could inline it, together with that script that joins it with the list of impacted modules, in the `ci.yml` file. But it seemed more cleaner put stuff into files.